File: app/src/main/java/at/shockbytes/coins/network/coinbase/CoinbasePriceApi.java
package at.shockbytes.coins.network.coinbase;

import at.shockbytes.coins.network.model.PriceConversion;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;
import rx.Observable;

/**
 * @author Martin Macheiner
 *         Date: 14.06.2017.
 */

public interface CoinbasePriceApi {

    String SERVICE_ENDPOINT = "https://api.coinbase.com/v2/prices/";

    @GET("{conversion}/spot")
    Observable<PriceConversion> getSpotPrice(@Path("conversion") String conversion,
                                @Header("CB-VERSION") String version);

    @GET("{conversion}/buy")
    Observable<PriceConversion> getBuyPrice(@Path("conversion") String conversion,
                                             @Header("CB-VERSION") String version);


    @GET("{conversion}/sell")
    Observable<PriceConversion> getSellPrice(@Path("conversion") String conversion,
                                            @Header("CB-VERSION") String version);

}

File: app/src/main/java/at/shockbytes/coins/network/coinbase/CoinbasePriceApi.kt
package at.shockbytes.coins.network.coinbase

import at.shockbytes.coins.currency.conversion.PriceConversion
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Path

/**
 * @author Martin Macheiner
 * Date: 14.06.2017.
 */

interface CoinbasePriceApi {

    @GET("{conversion}/spot")
    fun getSpotPrice(@Path("conversion") conversion: String,
                     @Header("CB-VERSION") version: String): Observable<PriceConversion>

    @GET("{conversion}/buy")
    fun getBuyPrice(@Path("conversion") conversion: String,
                    @Header("CB-VERSION") version: String): Observable<PriceConversion>


    @GET("{conversion}/sell")
    fun getSellPrice(@Path("conversion") conversion: String,
                     @Header("CB-VERSION") version: String): Observable<PriceConversion>

    companion object {

        val SERVICE_ENDPOINT = "https://api.coinbase.com/v2/prices/"
    }

}
